crates/core/src/commands/cat.rs

```diff
@@ -38,8 +40,17 @@ pub(crate) fn cat_file<P, S: Open>(
     repo: &Repository<P, S>,
     tpe: FileType,
     id: &str,
-) -> RusticResult<Bytes> {
-    let id = repo.dbe().find_id(tpe, id)?;
+) -> RusticResult<Bytes>
+where
+    P: Progress,
+{
+    // For SnapshotFile, using `SnapshotFile::from_str` allows `latest~N` syntax
+    let id = if let FileType::Snapshot = tpe {
+        let snap_file = SnapshotFile::from_str(repo.dbe(), id, |_| true, &repo.pb)?;
+        snap_file.id.into_inner()
+    } else {
+        repo.dbe().find_id(tpe, id)?
+    };
```

Comment:
Actually I must say I don't like the `cat_file` API as it is currently. I would now rather change it using `RepoId` and let the caller find out which is the correct `Id` - using a suitable search method on `Repository`, if needed.

The resolving of SnapshotIds would then be also part of such suitable search method...

As adding `P: Progress` would be also a breaking change, what do you think about my proposal?